src/screens/shared/dashboard/StatCards.js

```diff
@@ -22,49 +35,175 @@ const styles = {
     textAlign: 'center'
   }
 };
-class StatCards extends React.PureComponent {
-  getCardStats = () => {
-    // TODO: Replace with Airtable Call
+class StatCards extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      farms: [],
+      ksFarms: [],
+      totalHarvests: [],
+      recentHarvests: [],
+      gapCertification: [],
+      numFarmReferred: '',
+      numKsGapAccepted: '',
+      percentKsGapCertified: '',
+      percentKsGapApplied: '',
+      percentGapCertified: '',
+      nsevpHarvestFarms: '',
+      totalHarvestsPounds: '',
+      percentGapApplied: ''
+    };
+  }
+
+  async componentDidMount() {
+    const farms = await getAllFarmsForFarmSearch();
+    const gapCertification = await getAllGAPCertificationsForStatCard();
+    const totalHarvests = await getAllTotalHarvestsForStatCard();
+    const recentHarvests = await getAllRecentHarvestLogsForStatCard();
+    const ksFarms = farms.filter(farm => farm.ksAffiliated);
+
+    const numGapCertified = gapCertification.filter(farm => farm.gapCertified); // in gap certification there is no ks affiliated column
+
+    const numFarmReferred = gapCertification.filter(
+      farm => farm.farmReferred === 'Complete'
+    ).length;
+
+    const numKsGapAccepted = gapCertification.filter(
+      farm => farm.gapAccepted === 'Complete'
+    ).length;
+
+    const numGapApplied = gapCertification.filter(
+      farm => farm.farmApplied === 'Complete'
+    );
+
+    function roundToTwo(num) {
+      return +`${Math.round(`${num}e+2`)}e-2`;
+    }
+
+    function roundToOne(num) {
+      return +`${Math.round(`${num}e+1`)}e-1`;
+    }
+    const percentKsGapCertified = roundToOne(
+      (numGapCertified.length / ksFarms.length) * 100
+    );
+    const percentKsGapApplied = roundToTwo(
+      (numGapApplied.length / ksFarms.length) * 100
+    );
+
+    const percentGapCertified = roundToTwo(
+      (numGapCertified.length / farms.length) * 100
+    );
+    const percentGapApplied = roundToTwo(
+      (numGapApplied.length / farms.length) * 100
+    );
+    const nsevpHarvestFarms = recentHarvests.length;
+
+    let total = 0;
+    for (let i = 0; i < totalHarvests.length; i += 1) {
+      total += totalHarvests[i].totalProductionPounds;
+    }
+    const totalHarvestsPounds = total;
```

Comment:
Could you declare `let totalHarvestsPounds` so you don't need the `total` variable?